src/functions/handlers/handleEvents.js

```diff
@@ -1,38 +1,27 @@
-import fs from "fs";
+import fs from "node:fs";
+import path from "node:path";
+import { createRequire } from "node:module";
+const require = createRequire(import.meta.url);
 import mongoose from "mongoose";
 
+const EVENTS_PATH = './src/events/';
+
+const isJs = (file = "") => file.endsWith(".js");
+const isDir = (dir) => fs.statSync(path.join(EVENTS_PATH, dir)).isDirectory();
+
 export default (client) => {
-  client.handleEvents = async () => {
-    const eventFolders = fs
-      .readdirSync(`./src/events`)
-      .filter((folder) => fs.statSync(`./src/events/${folder}`).isDirectory());
+  const methods = { client, mongo: mongoose };
+  client.handleEvents = () => {
+    const eventFolders = fs.readdirSync(EVENTS_PATH).filter(isDir);
     for (const folder of eventFolders) {
-      const eventFiles = fs
-        .readdirSync(`./src/events/${folder}`)
-        .filter((file) => file.endsWith(".js"));
-      switch (folder) {
-        case "client":
-          for (const file of eventFiles) {
-            const event = (await import(`../../events/${folder}/${file}`)).default;
-            if (event.once) {
-              client.once(event.name, (...args) => event.execute(...args, client));
-            } else {
-              client.on(event.name, (...args) => event.execute(...args, client));
-            }
-          }
-          break;
-        case "mongo":
-          for (const file of eventFiles) {
-            const event = (await import(`../../events/${folder}/${file}`)).default;
-            if (event.once) {
-              mongoose.connection.once(event.name, (...args) => event.execute(...args, client));
-            } else {
-              mongoose.connection.on(event.name, (...args) => event.execute(...args, client));
-            }
-          }
-          break;
-        default:
-          break;
+      const eventPath = path.join(EVENTS_PATH, folder);
+      const eventFiles = fs.readdirSync(eventPath).filter(isJs);
```

Comment:
I'm not sure we're getting the correct file path here (```../../events/...```). I will be glad if you can check it, because I can’t do this without tests.